processor/src/main/java/org/mapstruct/ap/internal/model/PropertyMapping.java

```diff
@@ -638,16 +639,33 @@ private Assignment forgeMapping(SourceRHS sourceRHS) {
             if ( sourceType.isPrimitive() || targetType.isPrimitive() ) {
                 return null;
             }
+
             String name = getName( sourceType, targetType );
-            ForgedMethod forgedMethod = new ForgedMethod(
-                name,
-                sourceType,
-                targetType,
-                method.getMapperConfiguration(),
-                method.getExecutable(),
-                method.getContextParameters(),
-                getForgedMethodHistory( sourceRHS )
-            );
+            List<Parameter> parameters = new ArrayList( method.getContextParameters() );
+            ForgedMethod forgedMethod;
+            if ( method.isUpdateMethod() ) {
+                parameters.add( Parameter.forForgedMappingTarget( targetType ) );
```

Comment:
I would just add the new parameter if the `method` is update method. Everything else can stay as it was before. 

If the `mappingTargetParameter` is not `null` and it is when we do an update method, then we return it's type. The `TypeKind.VOID` is obsolete